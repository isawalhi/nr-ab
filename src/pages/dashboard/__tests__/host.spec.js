import Host from '../components/host/hostModel'
import HostView from '../components/host/hostView'
import Application from '../components/application/applicationModel';

describe('Host', () => {
    let host;
    const data = {
        name: 'test-app',
        contributors: ['test-cont-1'],
        version: 1,
        apdex: 89,
    }

    const app1 = new Application(data);
    const app2 = new Application({ ...data, name: 'test-app2', apdex: 99 })
    const app3 = new Application({ ...data, name: 'test-app3', apdex: 99 })

    const applications = [app1, app2]
    const hostName = 'test-host'

    beforeEach(() => {
        host = new Host(hostName, [...applications]);
    });

    it('assigns constructor values correctly', () => {
        expect(host.hostName).toBe(hostName)
        expect(host.applications).toStrictEqual(applications)
    })

    it('sets and gets applications correctly', () => {
        host.setApplications([app1])
        expect(host.getApplications()).toStrictEqual([app1])
    })

    it('adds new application correctly', () => {
        host.addApplication(app3)

        expect(host.getApplications()).toStrictEqual([app1, app2, app3])
    })

    it('removes application correctly', () => {
        host.removeApplication(app1)

        expect(host.getApplications()).toStrictEqual([app2])
    })

    it('sorts application correctly', () => {
        host.addApplication(app3)
        expect(host.sortApplications()).toStrictEqual([app2, app3, app1])
    })

    it('gets the first top sorted application when top is defined', () => {
        host.addApplication(app3)
        expect(host.sortApplications(2)).toStrictEqual([app2, app3])
    })

    it('gets the top sorted apps', () => {
        host.addApplication(app3)
        expect(host.getTopApplications()).toStrictEqual([app2, app3, app1])
    })

    it('generates html when host is provided correctly', () => {
        expect(new HostView().generateMarkup(host)).toMatchSnapshot()
    })
})