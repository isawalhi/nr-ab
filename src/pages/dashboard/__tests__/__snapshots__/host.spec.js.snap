// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Host generates html when host is provided correctly 1`] = `
"
        <div id=test-host class=\\"host-card\\">
          <div class=\\"host-name\\">test-host</div>
          <div class=\\"applications\\">
            
        <div key=\\"test-app2\\" release-version=\\"1\\" class=\\"application\\">
            <div class=\\"app-apdex\\">99</div>
            <div class=\\"app-name\\">test-app2</div>
        </div>
        
        <div key=\\"test-app\\" release-version=\\"1\\" class=\\"application\\">
            <div class=\\"app-apdex\\">89</div>
            <div class=\\"app-name\\">test-app</div>
        </div>
        
          </div>
        </div>
        "
`;
